### PR TITLE
rbd: output notifyOp request name when watching

### PR DESCRIPTION
**Pull request by shun-s**
output notifyOp request name when watching,
  which is more easier to dentify what happend.

Signed-off-by: shun-s <song.shun3@zte.com.cn>

### CODE REVIEW

File: src/librbd/WatchNotifyTypes.h [hunk 1]
```diff
@@ -339,6 +339,7 @@ struct NotifyMessage {
   void encode(bufferlist& bl) const;
   void decode(bufferlist::iterator& it);
   void dump(Formatter *f) const;
+  NotifyOp getOp() const;
```
Comment (dillaman) [inline src/librbd/WatchNotifyTypes.h]: ```get_notify_op```

File: src/librbd/WatchNotifyTypes.cc [hunk 2]
```diff
@@ -21,6 +21,14 @@ class CheckForRefreshVisitor  : public boost::static_visitor<bool> {
   }
 };
 
+class GetNotifyOPVisitor  : public boost::static_visitor<int> {
```
Comment (dillaman) [inline src/librbd/WatchNotifyTypes.cc]: Nit: ```GetNotifyOpVisitor```

Comment (dillaman) [inline src/librbd/WatchNotifyTypes.cc]: Nit: ```boost::static_visitor<NotifyOp>```

File: src/librbd/WatchNotifyTypes.cc [hunk 3]
```diff
@@ -373,6 +381,11 @@ void NotifyMessage::dump(Formatter *f) const {
   apply_visitor(DumpPayloadVisitor(f), payload);
 }
 
+NotifyOp NotifyMessage::getOp() const {
+  apply_visitor(GetNotifyOPVisitor(), payload);
+}
+
+
```
Comment (dillaman) [inline src/librbd/WatchNotifyTypes.cc]: Nit: remove extra blank line

File: src/tools/rbd/action/Watch.cc [hunk 4]
```diff
@@ -30,9 +31,25 @@ class RbdWatchCtx : public librados::WatchCtx2 {
...
+    using namespace librbd::watch_notify;
+    NotifyMessage notify_message;
+    if (bl.length() == 0) {
+      notify_message = NotifyMessage(HeaderUpdatePayload());
+    } else {
+      try {
+        bufferlist::iterator iter = bl.begin();
+        notify_message.decode(iter);
+      } catch (const buffer::error &err) {
+        std::cout << __func__ << " error decoding image notification: "
```
Comment (dillaman) [inline src/tools/rbd/action/Watch.cc]: Nit: ```std::cerr << "rbd: failed to decode image notification" << std::endl;```

File: src/tools/rbd/action/Watch.cc [hunk 5]
```diff
@@ -30,9 +31,25 @@ class RbdWatchCtx : public librados::WatchCtx2 {
...
+                              << err.what() << std::endl;
+        return;
```
Comment (dillaman) [inline src/tools/rbd/action/Watch.cc]: ... no need to return since it will default to unknown payload.

**Comment by shun-s** [comment]
@dillaman  all your requirements are done, please take a look again.

**Pull request merged**